test/module/shared_model/backend_proto/shared_proto_tx_response_test.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "backend/protobuf/transaction_responses/proto_tx_response.hpp"
+
+#include <gtest/gtest.h>
+
+// Quite hacky way to extract the class name
+// probably not stable and should be improved
+class ClassNameVisitor : public boost::static_visitor<const std::string> {
```

Comment:
I feel this class will be redundant when you implement `toString` method in responses.

Reply:
Yeah, you right. Just need to be moved later

Reply:
This visitor is redundant at all. Pls, check https://github.com/hyperledger/iroha/pull/720/files#diff-d2d707e60db84874bbaf915d2ec607ec Andrey's PR where he made the fine solution for this.